### ISSUE TITLE
TU not working

### ISSUE THREAD

**Issue by Djassie18698** [emoose/xenia#30]
I used an older canary first to use TU but saves were not working, so i tried this version. However it doesnt even read my TU when launching the game, do you know why maybe? It's about diablo 3

[//]: # (Describe what's going wrong:)

[//]: # (Describe what should happen:)

[//]: # (If applicable, provide a callstack here - esp. for crashes)

[//]: # (If applicable, upload a logfile and link it here)

**Comment by Permdog99** [comment]
Did you use stfs-writer?

**Comment by Djassie18698** [comment]
Yeah I did, however when launching diablo 3 it's not the updated version, and it doesnt save either, if i quit game then i need to start all over. I'm not seeing any new folders and that's weird because it can save with the newer canary version but then I can't use TU

**Comment by Djassie18698** [comment]
I mean new folders as in I deleted the xenia folder so it could remake itself but nothing's happening

**Comment by Permdog99** [comment]
I have had this issue with canary and it works fine with stfs-writer. stfs-writer does not work with canary saves. Make sure you are using stfs-writer by looking at your window bar. It should say "xenia (stfs-writer/"

**Comment by Djassie18698** [comment]
![image](https://user-images.githubusercontent.com/51295582/120902885-232a1100-c643-11eb-8ed5-7c5849ea5da8.png)
Yeah thats what im using but after i enter the village it saves, but if i quit and restart it's all gone

**Comment by Permdog99** [comment]
Try making a new save to see if that fixes it.

**Comment by Djassie18698** [comment]
![image](https://user-images.githubusercontent.com/51295582/120903040-080bd100-c644-11eb-8d41-a42e888a7472.png)
these are the folders, i will delete them all and try again

**Comment by Permdog99** [comment]
Emoose said he was gonna upload 1.0.6 soon and that it should fix the title updates for Minecraft. It may work for Diablo 3. I don't own the game so I cannot check.

**Comment by Permdog99** [comment]
> ![image](https://user-images.githubusercontent.com/51295582/120903040-080bd100-c644-11eb-8d41-a42e888a7472.png)
> these are the folders, i will delete them all and try again

The saves are in 0000001 or something. They should NOT be extracted. Canary extracts them when being used. stfs-writer can't use canary saves.

**Comment by Djassie18698** [comment]
What kind of files do they use? And are they just placed in 0000001?

**Comment by Permdog99** [comment]
Usually .bin and it is placed in 00000001.

**Comment by Djassie18698** [comment]
![image](https://user-images.githubusercontent.com/51295582/120905000-87eb6880-c64f-11eb-939b-ba8ded1a4ab4.png)
I played some diablo 3 but there is not even a new folder created, not sure why

**Comment by Permdog99** [comment]
Don't delete the content folder, delete the save that is inside of /content/[GameID]/00000001/ . Also do you have a portable.txt in the same folder as the xenia.exe?

**Comment by Djassie18698** [comment]
I dont have a portable in my xenia folder, should I add it?

**Comment by Permdog99** [comment]
Show me your xenia folder.

**Comment by Djassie18698** [comment]
Im playing diablo right now and still no new files and this is how it looks
![image](https://user-images.githubusercontent.com/51295582/120907131-70b47700-c65f-11eb-8c10-3843bbc16326.png)

**Comment by Permdog99** [comment]
Add portable.txt then try it.

**Comment by Djassie18698** [comment]
![image](https://user-images.githubusercontent.com/51295582/120907238-33041e00-c660-11eb-97ef-ab6e756595ee.png)
With portable it creates files. but no saves, and diablo said it saved already

**Comment by Permdog99** [comment]
Show me a video of your issue.

**Comment by Djassie18698** [comment]
There is not much to record, but when i launch diablo 3 i can play normally, however once it says "checkpoint reached" and close the game after i need to start again like there are no saves, and as you see I dont have a content folder added

**Comment by Permdog99** [comment]
Re download Xenia (stfs-writer) and use that one. You may just need to update.

**Comment by Djassie18698** [comment]
I already have the last version sadly

**Comment by Permdog99** [comment]
It updated last night.

**Comment by Djassie18698** [comment]
Yeah it's the on I have, i only downloaded it ~3 hours ago

**Comment by Permdog99** [comment]
Send me a video.

**Comment by Djassie18698** [comment]
What would you like to see? Because the problem is that there is nothing happening in the background :p

**Comment by Permdog99** [comment]
I want to see the issue so I can find a fix.

**Comment by Djassie18698** [comment]
I will send a video asap, but normally it creates a content folder but now it doesn't so if i launch the game again there are no saves. So im not sure of what i should make a video of, because there is not much to show you guys

**Comment by Djassie18698** [comment]
Where did you download 1.0.6? Since the only version i see is a 1.0.5.zip

**Comment by Permdog99** [comment]
Nvm that was just the version of Minecraft.

**Comment by emoose** [comment]
1.0.7 fixes issues with TUs not applying to ISO/disc-originated games, and should hopefully fix the 35GB big-package bug.

**Comment by GinjaNinja1998** [comment]
emoose, this is the xex and the TU you have requested
[Title Update and XEX.zip](https://github.com/emoose/xenia/files/6641628/Title.Update.and.XEX.zip)

**Comment by emoose** [comment]
@GinjaNinja1998 hm ok, seems I was looking at wrong TU, the one you posted is for the 22AB4F7C media-ID, seems it patches version 10.0 to 10.1.

Looking at the XEX you posted, it's already been patched to 10.1 though, was that done by you? Xenia can't patch an XEX that's already been patched.

But if the x360 accepts that pre-patched XEX & also uses the TU, I guess it's skipping the actual XEXP patching since it's already patched, and just makes use of the files available in the TU ("TU__GearCamera.ini" "TU__GearPawn.ini" etc), maybe need to look into some way for Xenia to do the same.

**Comment by GinjaNinja1998** [comment]
> @GinjaNinja1998 hm ok, seems I was looking at wrong TU, the one you posted is for the 22AB4F7C media-ID, seems it patches version 10.0 to 10.1.
> 
> Looking at the XEX you posted, it's already been patched to 10.1 though, was that done by you? Xenia can't patch an XEX that's already been patched.
> 
> But if the x360 accepts that pre-patched XEX & also uses the TU, I guess it's skipping the actual XEXP patching since it's already patched, and just makes use of the files available in the TU ("TU__GearCamera.ini" "TU__GearPawn.ini" etc), maybe need to look into some way for Xenia to do the same.

Yes I patched that particular XEX. I ran an unpatched xex with the TU in the documents folder, and ran a patched xex without the TU in the documents folder. Either way presented a crash when loading the game. For good measure, I even ran an unpatched god version with the TU in the documents folder too, and even that crashes.

**Comment by emoose** [comment]
Ah right, any chance you can upload the unpatched XEX somewhere too? I'll see if I can get it to trigger on my end, not sure if I'll be able to without the game files tho, but maybe I can just use files from a different gears 2 version.

If you can run xenia with log_level 3, then run the unpatched XEX with TU in documents, the xenia.log might have some more info about the crash inside as well, if you could post that too it might be useful.

**Comment by GinjaNinja1998** [comment]
Yes when I am off work I'll get that to you. that'll be in about 4ish hours or so

**Comment by GinjaNinja1998** [comment]
[Uploading log and xex.zip…]()
This should be the unpatched xex. Along with it is the log file

**Comment by emoose** [comment]
Well it looks like the TU is being applied at least, as you can see in that log "XEX patch applied successfully: base version: 10.0.0.0, new version: 10.0.1.0"

Game seems to run and try accessing a bunch of stuff from the TU, like "UPDATE:\TU__Xe-GearUI.ini", I'm guessing either the patched XEX, or something read from those TU UPDATE:\ files, is triggering some emu bug which makes it crash out.

If you try running the pre-patched XEX with the latest xenia-master (not stfs-writer/canary), does the game also crash or does it run fine? That might tell us if it's the XEX patch or the TU files that cause it, since master can't access UPDATE:\

Unfortunately crash cause doesn't seem to get logged, I'll have to try getting hold of Gears 2 and see if I can make that crash with a debugger.
